tests/unit/CommandTest.php

```diff
@@ -204,7 +204,7 @@ public function testBindParamValue()
         $row = $command->queryOne();
         $this->assertEquals($intCol, $row['int_col']);
         $this->assertEquals($charCol, $row['char_col']);
-        $this->assertEquals($floatCol, $row['float_col']);
+        $this->assertEquals((float)$floatCol, (float)$row['float_col']);
```

Comment:
Copied from #40: 

> If that test was copied over from yii2 and was working without casting to float, we better understand what changed and why.
>
> Could you check how it's done in yii2?

Reply:
The same test is present here for yii2:
   https://github.com/yiisoft/yii2/blob/master/tests/framework/db/CommandTest.php#L210

MySQL 5.7 is used for yii2: https://github.com/yiisoft/yii2/blob/master/.travis.yml#L55
(this can also be seen in the logs: )

Reply:
(You may want to compare data sets used for tests between yii2 & yii3)

Reply:
I found the problem.

Sebatian comparator package has changed this line:

https://github.com/sebastianbergmann/comparator/blob/33825cd25f073faafe98c4a017a5d2b69f3e09d3/src/ScalarComparator.php#L64

at this commit:

https://github.com/sebastianbergmann/comparator/commit/c4e6d5c693d4dce8f3db478f09ef058b1b9c987e

So, Yii2 does a "!=" check, instead Yii3 does a "!==" check, that fails, because db field is "1.230", instead floatval is "1.23".

So I think that my PR is valid.

Reply:
@FabrizioCaldarelli  This should not be the case, since `$floatCol` should be either float or int in this test.

Reply:
@rob006 Exactly, $floatCol is a float (1.23), but $row['float_col'] is a string, returned by $query->one(), with "1.230" value.

Since Sebastian comparator packages changed check from "!=" to "!==", now the two values don't match.

Reply:
@FabrizioCaldarelli But [this condition](https://github.com/sebastianbergmann/comparator/blob/33825cd25f073faafe98c4a017a5d2b69f3e09d3/src/ScalarComparator.php#L64) will pass only if both arguments are strings, so not in this case. And then [this condition](https://github.com/sebastianbergmann/comparator/blob/33825cd25f073faafe98c4a017a5d2b69f3e09d3/src/ScalarComparator.php#L75) will be checked, but it should return `false`: https://3v4l.org/eui82 . 

So definitely something weird is happening here.

Reply:
@rob006 This is caused by PDOStatement::bindParam. At beginning $floatVal is a float, but after executing bindParam: 
https://github.com/yiisoft/db/blob/54e7a7ef9f7683bb641a13574c5c34cbd8c5dc61/src/Command.php#L298

PDOStatement::bindParam will convert numeric value in string, as described here: https://github.com/yiisoft/db/blob/54e7a7ef9f7683bb641a13574c5c34cbd8c5dc61/src/Command.php#L298